core/templates/components/skill-selector/skill-selector.component.ts

```diff
@@ -194,9 +195,14 @@ export class SkillSelectorComponent implements OnInit {
   }
 
   searchInUntriagedSkillSummaries(searchText: string): SkillSummary[] {
-    let skills: string[] = this.untriagedSkillSummaries.map(val => {
-      return val.description;
-    });
+    let skills: string[] = this.untriagedSkillSummaries
+      .filter(
+        val => !this.addedSkillsIds ? true :
+        !this.addedSkillsIds.includes(val.id)
```

Comment:
Can you use a data structure for addedSkillIds (or rather, skillIdsToExclude) that supports lookup of O(1) rather than O(N)?

E.g. a dict mapping each excluded skill ID to true, and you use hasOwnProperty to check membership.

Reply:
done